assets/javascripts/discourse/models/docs.js

```diff
@@ -38,6 +39,11 @@ Docs.reopenClass({
     }
 
     return ajax(`/${docsPath}.json?${filters.join("&")}`).then((data) => {
+      const site = Site.current();
+      data.categories?.forEach((category) => site.updateCategory(category));
+      data.topics.topic_list.categories?.forEach((category) =>
+        site.updateCategory(category)
```

Comment:
I'm wondering if we should be expecting plugins to have to call `updateCategory`. From a plugin developer's perspective, the reason they have to do this might not make sense to them. Is there a downside in dropping this code?

Reply:
I think we cannot avoid calling `site.updateCategory`, but maybe we can avoid plugin developers from doing that sometimes. For example, in this case, if we used code for managing topic lists from core, that already calls `site.updateCategory`, it would not be necessary to do it here anymore.